homeassistant/components/switch/rainbird.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for Rain Bird Irrigation system LNK WiFi Module.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/rainbird/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import (CONF_PLATFORM, CONF_SWITCHES, CONF_ZONE,
+                                 CONF_FRIENDLY_NAME, CONF_TRIGGER_TIME,
+                                 CONF_SCAN_INTERVAL, CONF_HOST, CONF_PASSWORD)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.exceptions import PlatformNotReady
+
+REQUIREMENTS = ['pyrainbird==0.0.9']
+
+DOMAIN = 'rainbird'
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): DOMAIN,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_SWITCHES, default={}): vol.Schema({
+        cv.string: {
+            vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+            vol.Required(CONF_ZONE): cv.string,
+            vol.Required(CONF_TRIGGER_TIME): cv.string,
+            vol.Optional(CONF_SCAN_INTERVAL): cv.string,
+        },
+    }),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Rain Bird switches over a Rain Bird controller."""
+    server = config.get(CONF_HOST)
+    password = config.get(CONF_PASSWORD)
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
```

Comment:
You are not allowed to use the entity component. This is a platform for the switch entity component.